### PR TITLE
[Translate ENG-KO] updates/2016/05/Houdini.md

### PR DESCRIPTION
**Pull request by jihyerish**
I've translated the updates about houdini APIs.

### CODE REVIEW

**Comment by jihyerish** [comment]
Hi, @cwdoh. 
I've found you in the list of WebFundamentals moderators.
This is the first time for me to contribute WebFundamentals. 
So I'm wondering how to proceed with this PR, so I've contacted you.
Could you give me some guideline and review this PR?

Thanks in advance! : )

**Comment by ragingwind** [comment]
Great job! @jihyerish I guess you can try the translation with this https://developers.google.com/web/resources/translations But please considering that it is in the experimental stage yet.

**Comment by jihyerish** [comment]
@ragingwind Thanks for the guideline! Using gitlocalize seems really convenient!
Then do I need to close this PR and start on the gitlocalize?

**Comment by jenix21** [comment]
@jihyerish @ragingwind 
This pr is on progress?

**Comment by jihyerish** [comment]
@jenix21 I guess this PR is waiting for the review.

**Comment by jenix21** [comment]
Thanks @jihyerish 
Reviews at https://gitlocalize.com/repo/107/ko/review appear to be not that active.

**Comment by jihyerish** [comment]
I've requested the review on https://gitlocalize.com/repo/107/ko/review/8482.

**Comment by ragingwind** [comment]
@jihyerish I will let reviewers to take on this. please understand that it's not so very activated yet.

**Comment by jihyerish** [comment]
@ragingwind Sure. :) Thank you for your support!

File: src/content/ko/updates/2016/05/houdini.md [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+ 변경할 수 있습니다. 그런 점에 있어서 *마법*과도 같죠. (앞으로 제가 무슨 말을 할 지 감이 오시나요?!) 지금까지 우리 웹 개발자들은
+이러한 마법을 바라보기만 했었습니다. 만약 여러분만의 마법을 부리고 싶다면요? 만약 우리가 *마법사가 되고 싶다면요*?
+Houdini를 시작해보세요!
+
+Houdini 프로젝트는 Mozilla, Apple, Opera, Microsoft, HP, Intel 그리고 Google의 엔지니어들로 구성되어 (번역자: LG도 있습니다!)
+CSS 엔진의 일부분에 웹 개발자들이 접근할 수 있도록 협업하고 있습니다. 이 프로젝트는 공식 W3C 표준으로 채택되기 위한
+*표준 초안*들을 작성하고 있습니다. 프로젝트의 구성원들은 표준안에 대한 몇 가지 높은 수준의 목표를 세워 표준 초안을 작성하였으며 이를 기반으로
+하위 수준 사양의 표준 문서들이 개발되었습니다. 결국, “Houdini”라하면 이 표준 문서들의 내용을 의미합니다.
+표준안 개발 작업이 진행되는 동안 [Houdini 표준안 초안][Houdini Drafts]들은 미완성 단계이며 일부 초안은 다소 가안입니다.
+이것이 Houdini를 개발하는 초기 단계입니다.
```
Comment (cwdoh) [inline src/content/ko/updates/2016/05/houdini.md]: 문장을 좀 의역해보는게 어떨까요?
> 지금까지 Houdini의 초기 개발 단계가 어떻게 진행되고 있는지 설명해드렸습니다.

File: src/content/ko/updates/2016/05/houdini.md [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+CSS가 할 수 있는 일들을 생각해본 적이 있나요? 여러분이 하나의 CSS 속성을 바꾸면 전체 웹 페이지를 완전히 다른 레이아웃으로
+ 변경할 수 있습니다. 그런 점에 있어서 *마법*과도 같죠. (앞으로 제가 무슨 말을 할 지 감이 오시나요?!) 지금까지 우리 웹 개발자들은
+이러한 마법을 바라보기만 했었습니다. 만약 여러분만의 마법을 부리고 싶다면요? 만약 우리가 *마법사가 되고 싶다면요*?
+Houdini를 시작해보세요!
+
+Houdini 프로젝트는 Mozilla, Apple, Opera, Microsoft, HP, Intel 그리고 Google의 엔지니어들로 구성되어 (번역자: LG도 있습니다!)
+CSS 엔진의 일부분에 웹 개발자들이 접근할 수 있도록 협업하고 있습니다. 이 프로젝트는 공식 W3C 표준으로 채택되기 위한
+*표준 초안*들을 작성하고 있습니다. 프로젝트의 구성원들은 표준안에 대한 몇 가지 높은 수준의 목표를 세워 표준 초안을 작성하였으며 이를 기반으로
+하위 수준 사양의 표준 문서들이 개발되었습니다. 결국, “Houdini”라하면 이 표준 문서들의 내용을 의미합니다.
+표준안 개발 작업이 진행되는 동안 [Houdini 표준안 초안][Houdini Drafts]들은 미완성 단계이며 일부 초안은 다소 가안입니다.
```
Comment (cwdoh) [inline src/content/ko/updates/2016/05/houdini.md]: 제안
> 표준안 초안 -> 초안
> ... 완료되지 않았으며, 초안 중 일부는 작업 예정 사항입니다.

File: src/content/ko/updates/2016/05/houdini.md [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+주변 시스템에 연결할 수 있도록 새로운 API를 제공하는 것입니다.  아마도 이러한 코드의 일부를 모든 단일 프레임에서
+실행한다고 가정해보는 것은 비현실적이지는 않습니다. 어쩌면 그렇게 되는 것이 당연합니다.
+[Web Worker 표준]에는 다음과 같은 내용이 있습니다.:
+
+> Workers [...] 는 상대적으로 무거우며 한 페이지에 많은 수의 Workers 사용은 권장하지 않습니다. 예를 들어 4 mega pixel
+이미지의 각 pixel마다 하나의 Worker를 생성하는 것은 적절하지 않습니다.
+
+이는 곧 web workers가 Houdini가 하려고 했던 것을 하기에는 무리라는 의미입니다. 따라서 worklets이 만들어졌습니다.
+Worklets은 ES2015의 class들을 써서 method 집합들, worklet의 미리 정의된 타입인 signature들을 정의합니다.
+그렇기 때문에 Worklet은 보다 경량화가 되었으며 수명 주기가 짧습니다.
```
Comment (cwdoh) [inline src/content/ko/updates/2016/05/houdini.md]: 보통 [생명 주기](http://terms.tta.or.kr/mobile/dictionaryView.do?word_seq=047526-1)를 좀 더 많이 사용합니다.

File: src/content/ko/updates/2016/05/houdini.md [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+
+**Status update:** Paint API는 Chrome 65의 기본 기능으로 제공됩니다. 자세한 내용은 [여기](/web/updates/2018/01/paintapi)를 참고해주세요.
+
+### Compositor worklet
+
+참고: 이 곳에 서술된 API는 더 이상 사용되지 않습니다. Compositor worklet 는 현재 “Animation Worklet”으로
+새로 디자인되었습니다. 해당 API에 대한 더 자세한 사항은
+[여기](https://dassur.ma/things/animworklet/)에서 찾아보실 수 있습니다.
+
+Compositor Worklet이 WICG로 옮겨졌고 계속 내용이 바뀌고 있지만, 저에게는 가장 흥미로운 표준안입니다. 아시다시피 일부 작업은 CSS 엔진에 의해 그래픽 카드로 아웃소싱되며 일반적으로 그래픽 카드와 장치 모두에 의존합니다. 브라우저는 일반적으로 DOM 트리를 사용하고 특정 기준에 따라 일부 branch 혹은 하위 트리에 자체 [레이어][HTML5Rocks layers]를 공개합니다.
```
Comment (cwdoh) [inline src/content/ko/updates/2016/05/houdini.md]: `저에게는`는 번역 시 제외하시는게 더 자연스러울 것 같습니다. :)

File: src/content/ko/updates/2016/05/houdini.md [hunk 5]
```diff
@@ -0,0 +1,265 @@
...
+이러한 하위 트리는 스스로 해당 레이어에 그립니다. (향후에는 paint worklet을 사용하여 그릴 수도 있겠죠!) 마지막 단계로, 페인트 과정이 완료된 모든 개별 레이어는 z-index, 3D Transform등을 고려하여 서로 중첩이 되어 화면에 보이는 최종 이미지를 완성합니다. 이 프로세스를 “컴포지팅 (Compositing)”이라고 하며 컴포지터(Compositor)가 실행합니다. 이 프로세스의 장점은 페이지가 조금 스크롤될 때 *모든* 요소를 다시 그릴 필요가 없다는 것입니다. 대신, 이전 프레임에서 레이어를 재사용하고 변경된 스크롤 위치로 컴포지터를 다시 실행할 수 있습니다. 이로 인하여 빠른 처리가 가능합니다. 이는 60fps를 맞출수 있게 해줍니다. 이런 성능 개선은 [Paul Lewis]를 행복하게 해줍니다!
```
Comment (cwdoh) [inline src/content/ko/updates/2016/05/houdini.md]: 제안
> 해줍니다. -> 해줄 것입니다.

File: src/content/ko/updates/2016/05/houdini.md [hunk 6]
```diff
@@ -0,0 +1,265 @@
...
+    });
+
+
+저의 동료 Robert Flack이 Compositor Worklet을 위한 [polyfill][CompWorklet polyfill]을
+구현했으니 사용해보세요 - 확실히 성능이 향상된 것을 볼 수 있습니다.
+
+### Layout worklet
+([spec][Layout Worklet spec])
+
+참고: 첫 공식 표준안이 제안되었음 표준안의 구현도 상당히 진행되었습니다.
```
Comment (cwdoh) [inline src/content/ko/updates/2016/05/houdini.md]: `... 제안되었으며, 표준안의 구현도...`와 같이 원문을 조금 바꾸셔도 좋을 듯 합니다.

File: src/content/ko/updates/2016/05/houdini.md [hunk 7]
```diff
@@ -0,0 +1,265 @@
...
+
+다시 말하지만, 본 표준안의 실질적인 내용은 그리 많지는 않으나, 개념은 흥미롭습니다.: 여러분만의 레이아웃을 작성해보세요!
+Layout Worklet은 여러분이 `display: layout('myLayout')`와 같은 자바스크립트를 코드를 작성하면
+이를 실행하여 구현 내용에 따라 특정 노드의 box안에 해당 노드의 자식 요소들을 배치합니다.
+물론, CSS `flex-box` Layout을 모두 자바스크립트로 구현하는 것은 동일한 기능을 native에서
+실행하는 것보다 느리겠지만 성능 상의 이득을 얻을 수 있는 지름길을 택하는 시나리오를 생각해볼 수 있습니다.
+ Windows 10 또는  [Masonry] 스타일의 레이아웃으로 구성된 웹 페이지를 상상해보세요. absolute/fixed positioning은 사용되지 않으며,
+`z-index` 나 요소가 겹치거나 어떤 종류의 border 혹은 overflow도 없습니다. 재 레이아웃 시 이러한 모든 검사를 건너뛸 수 있으면 성능이 향상될 수 있습니다.
```
Comment (cwdoh) [inline src/content/ko/updates/2016/05/houdini.md]: `재 레이아웃`에 `(re-layout)`을 추가해주시면 더 좋을 것 같습니다. :)

File: src/content/ko/updates/2016/05/houdini.md [hunk 8]
```diff
@@ -0,0 +1,265 @@
...
+    $('#div3').styleMap.set('background-size',
+      [new SimpleLength(200, 'px'), w1.add(w2)])
+    $('#div4')).styleMap.get('margin-left')
+      // => {em: 5, percent: 50}
+
+
+### Properties and values
+([spec][Properties and Values spec])
+
+참고: 표준안은 상당히 완성되었습니다. 하지만 아직 사용 가능한 구현은 없습니다.
```
Comment (cwdoh) [inline src/content/ko/updates/2016/05/houdini.md]: 제안
> ... 되었습니다만, 아직 ...

File: src/content/ko/updates/2016/05/houdini.md [hunk 9]
```diff
@@ -0,0 +1,265 @@
...
+
+[CSS Custom Properties] (비공식적으로는 “CSS Variables”로도 불립니다.) 를 아시나요?
+이것은 타입이 있는 변수입니다! 지금까지, 변수들은 string 값들 만을 가지거나
```
Comment (cwdoh) [inline src/content/ko/updates/2016/05/houdini.md]: `들 만`으로 띄어쓰기가 되어 있습니다.

**Comment by ragingwind** [comment]
@cwdoh @jihyerish already using gitlocalize, https://gitlocalize.com/repo/107/ko/review/8482

**Comment by jihyerish** [comment]
> @jihyerish 개인적인 사정으로 한동안 WebFundamentals 프로젝트에 신경을 쓰지 못했습니다. 늦었지만 PR에 대한 리뷰를 추가하였으니, 한번 살펴봐주시면 고맙겠습니다. 번역하시느라 정말 수고하셨습니다. :)
> 
> Sorry for my late review on your PR, and also just wanna say Thank you for great work. I addressed some nits on the translation, but there's no error to understand meaning.

우와 정말 감사드립니다!! 리뷰해주신 내용 여기에서 바로 수정하면 될까요?
혹은 gitlocalize로 고치면 될까요?
WebFundamentals 번역 처음해보아서 절차가 어떻게 되는지 궁금합니다.

**Comment by cwdoh** [comment]
@jihyerish 어디에 작업하셔도 괜찮습니다.

@ragingwind I know, but I'm confused to review PR with GitLocalize. I think
I've been away from this project for so long.

2019년 4월 1일 (월) 오전 10:37, Jihye Hong <notifications@github.com>님이 작성:

**Comment by ragingwind** [comment]
@cwdoh I thought that gitlocalize is the platform we should use. @agektmr What are your thought on this?

**Comment by agektmr** [comment]
@ragingwind Ideally using GitLocalize would help. But it's flexible enough to do it otherwise. We are working closely with GitLocalize team to improve your translation experience. Stay tuned. @cwdoh

**Comment by cwdoh** [comment]
LGTM. I'll be back soon with gitlocalize :)

**Comment by agektmr** [comment]
@cwdoh @ragingwind There still remains some comments to resolve. Do you think we are good to go?

**Comment by ragingwind** [comment]
@agektmr I thought that some of unresolves are minor issue and we can keep doing it on gitlocalize.

**Comment by agektmr** [comment]
@ragingwind ok, then I'll wait for LGTM from @cwdoh .

**Comment by jihyerish** [comment]
Oh, I just see the comments above. Thank you for taking care of this PR!
I'll continue working this via gitlocalize!

**Comment by agektmr** [comment]
@jihyerish @ragingwind @cwdoh Thank you for your contribution, all!

**Pull request merged**